src/components/StarRating.spec.js

```diff
@@ -0,0 +1,24 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import StarRating from './StarRating';
+
+describe('StarRating', () => {
+  it('renders a heading and 5 buttons for rating', () => {
+    render(<StarRating />);
+
+    const heading = screen.getByText(/rate/i);
+    expect(heading).toBeInTheDocument();
+
+    const ratingButtons = screen.getAllByRole('button');
+    expect(ratingButtons).toHaveLength(5);
+  });
+
+  it('calls a function to set rating when clicking a button', () => {
+    const ratingCallback = jest.fn();
+    render(<StarRating saveRatingToDrink={ratingCallback} />);
+
+    const ratingButtons = screen.getAllByRole('button');
+    userEvent.click(ratingButtons[0]);
+    expect(ratingCallback).toHaveBeenCalled();
```

Comment:
It would be better to check if the callback is called with the index, to check if this is called with the correct rating. :)

Reply:
I have changed this, thank you :)